### PR TITLE
feat(elements): OTP Improvements

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

This PR ensures that the following things work correctly with our OTP input:

- The 1Password/Password manager icon is on the side of the input
- You can use Left/Right arrow to navigate
- You can use Home/End or Alt+Left or Alt+Right to get to Start/End
- You can use Shift + Left/Right to mark section
- You can use shift + A or double click to select all
- You can use paste on iOS/iPhone to paste OTP codes

Some things on that list already worked correctly (in some cases) but now it should be more consistent.

## Checklist

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

### CODE REVIEW

File: packages/elements/src/react/common/form/otp.tsx [hunk 1]
```diff
@@ -12,29 +13,66 @@ export type OTPInputProps = Exclude<
...
+      length={length}
+      autoSubmit={autoSubmit}
+    />
+  );
+});
+
+/**
+ * Standard `<input />` element that receives the same props as the OTPInput component except for the render prop.
+ */
+const OTPInputStandard = React.forwardRef<HTMLInputElement, Omit<OTPInputProps, 'render'>>(function OTPInput(
```
Comment (LekoArts) [inline packages/elements/src/react/common/form/otp.tsx:54]: I moved the standard component from an inline definition in `OTPInput` to its own component. Mostly to support the `autoSubmit` feature.

File: packages/elements/src/react/common/form/otp.tsx [hunk 2]
```diff
@@ -45,120 +83,314 @@ export const OTPInput = React.forwardRef<HTMLInputElement, OTPInputProps>(functi
...
+
+      const originalSelect = element.select.bind(element);
+      element.select = () => {
+        originalSelect();
+        setSelectionRange([0, element.value.length]);
+      };
+    }, []);
+
+    // Continuously update the selection range when the input is focused
+    React.useEffect(() => {
```
Comment (LekoArts) [inline packages/elements/src/react/common/form/otp.tsx:118]: This replaces the previous useLayoutEffect effect

File: packages/elements/src/react/common/form/otp.tsx [hunk 3]
```diff
@@ -45,120 +83,314 @@ export const OTPInput = React.forwardRef<HTMLInputElement, OTPInputProps>(functi
...
 
     return (
       <div
-        style={
-          {
-            position: 'relative',
-          } as React.CSSProperties
-        }
+        data-otp-input-segmented-wrapper
+        style={wrapperStyle}
```
Comment (LekoArts) [inline packages/elements/src/react/common/form/otp.tsx:158]: Moved the styles from inline to dedicated objects (throughout). Also added data attributes where applicable to allow for CSS styling

File: packages/elements/src/react/common/form/otp.tsx [hunk 4]
```diff
@@ -45,120 +83,314 @@ export const OTPInput = React.forwardRef<HTMLInputElement, OTPInputProps>(functi
...
-                  selectionRange[0] === selectionRange[1] && selectionRange[0] === i
-                    ? 'cursor'
-                    : selectionRange[0] <= i && selectionRange[1] > i
-                    ? 'selected'
-                    : 'none',
-                index: i,
-              })}
-            </React.Fragment>
-          ))}
+          {Array.from({ length: length }).map((_, i) => {
```
Comment (LekoArts) [inline packages/elements/src/react/common/form/otp.tsx:238]: Cleaned up the inline ternary here and added `isFocused` to the logic. This replaces the previous `onBlur` handler that reset the selection when the input was not focused anymore.

**Review by LekoArts — COMMENTED** [review]
Comments

**Review by tmilewski — APPROVED** [review]

**Pull request merged**